apps/src/templates/CodeWorkspaceContainer.jsx

```diff
@@ -5,13 +5,68 @@
 
 import $ from 'jquery';
 import React, {PropTypes} from 'react';
-var ReactDOM = require('react-dom');
-var Radium = require('radium');
-var utils = require('../utils');
-var commonStyles = require('../commonStyles');
+import ReactDOM from 'react-dom';
+import Radium from 'radium';
 import { connect } from 'react-redux';
+import * as utils from '../utils';
+import commonStyles from '../commonStyles';
 
-var styles = {
+class CodeWorkspaceContainer extends React.Component {
+  static propTypes = {
+    // redux provided
+    hidden: PropTypes.bool.isRequired,
+    isRtl: PropTypes.bool.isRequired,
+    noVisualization: PropTypes.bool.isRequired,
+
+    // not in redux
+    topMargin: PropTypes.number.isRequired,
+    children: PropTypes.node,
+  };
+
+  /**
+   * Called externally
+   * @returns {number} The height of the rendered contents in pixels
+   */
+  getRenderedHeight() {
+    return $(ReactDOM.findDOMNode(this)).height();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.topMargin !== prevProps.topMargin) {
+      utils.fireResizeEvent();
+    }
+  }
+
+  render() {
+    const {hidden, isRtl, noVisualization, topMargin, children} = this.props;
+    const mainStyle = {
+      ...styles.main,
+      top: topMargin,
+      ...(noVisualization && styles.noVisualization),
+      ...(isRtl && styles.mainRtl),
+      ...(noVisualization && isRtl && styles.noVisualizationRtl),
+      ...(hidden && commonStyles.hidden),
+    };
+
+    return (
+      <div style={mainStyle} className="editor-column">
+        <div id="codeWorkspace" style={styles.codeWorkspace}>
+          {children}
+        </div>
+      </div>
+    );
+  }
+}
+
+export const UnwrappedCodeWorkspaceContainer = Radium(CodeWorkspaceContainer);
```

Comment:
What's the difference between Unwrapped and Unconnected? I'm just used to seeing Unconnected to indicate the component isn't connected to the Redux store.

Reply:
Just trying out different terminology.  `Unconnected` seems very redux-specific while `Unwrapped` seems more generic to me.  That said, this is not actually an unwrapped component - it's already wrapped with Radium.  Maybe I should just call it `TestableCodeWorkspaceContainer`?

Reply:
`TestableCodeWorkspaceContainer` makes sense to me. `Unwrapped` gave me pause because I was trying to figure out what it was otherwise wrapped in besides Radium.